Switch to system libtool

sidsbrmnn opened the pull request:
Run `libtoolize` as part of `./autogen` to regenerate libtool-related macros and scripts.

Additionally, add `.gitignore` to each directory that generates a distribution file to ensure nothing outside of the source code is adding to version control even by mistake.

chandraravipati commented:
@sidsbrmnn , please add more detailed description explaining the need for these changes and explaining the changes.

sidsbrmnn commented:
> @sidsbrmnn , please add more detailed description explaining the need for these changes and explaining the changes.

I've updated the description. I also updated the title to 'Untracking libtool-related files from Git' which is more apt to the changes made.

amandaTrusted commented:
Thank you for the contribution! We will be pulling this in the next release.
